examples/TransactionGenerator.hs

```diff
@@ -48,246 +66,484 @@ import Servant.API
+import Chainweb.HostAddress
+import Chainweb.Pact.RestAPI
+import Chainweb.Simulate.Contracts.Common
 import Chainweb.Simulate.Contracts.CryptoCritters
 import Chainweb.Simulate.Contracts.HelloWorld
 import Chainweb.Simulate.Contracts.SimplePayments
+import Chainweb.Simulate.Contracts.CoinContract
 import Chainweb.Simulate.Utils
-import Chainweb.Pact.RestAPI
 
-data TransactionCommand = NoOp | DeployContracts [FilePath] | RunStandardContracts | RunSimpleExpressions | PollRequestKeys [ByteString]
-  deriving (Show, Eq, Read, Generic)
+newtype MeasureTime = MeasureTime
+  { measureTime :: Bool
+  } deriving (Eq, Show, Generic)
+    deriving newtype (Read)
 
-instance FromJSON TransactionCommand
+instance FromJSON MeasureTime
 
-instance ToJSON TransactionCommand
+instance ToJSON MeasureTime
+
+instance Default MeasureTime where
+  def = MeasureTime False
 
-newtype ChainwebPort = ChainwebPort { _chainwebPort :: Int}
-  deriving (Eq, Generic)
-  deriving newtype Show
+data TransactionCommand
+  = DeployContracts [ContractName] MeasureTime
+  | RunStandardContracts TimingDistribution MeasureTime
+  | RunSimpleExpressions TimingDistribution MeasureTime
+  | PollRequestKeys [ByteString] MeasureTime
+  | ListenerRequestKey ByteString MeasureTime
+  deriving (Show, Eq, Generic)
 
-instance FromJSON ChainwebPort
+instance FromJSON TransactionCommand
+
+instance ToJSON TransactionCommand
 
-instance ToJSON ChainwebPort
+transactionCommandToText :: TransactionCommand -> Text
+transactionCommandToText = T.decodeUtf8 . transactionCommandBytes
+{-# INLINE transactionCommandToText #-}
+
+transactionCommandBytes :: TransactionCommand -> B8.ByteString
+transactionCommandBytes t =
+  case t of
+    DeployContracts contracts (MeasureTime mtime) ->
+      "deploy [" <>
+      B8.intercalate "," ((B8.pack . getContractName) <$> contracts) <>
+      "] " <>
+      (fromString . map toLower . show $ mtime)
+    RunStandardContracts d (MeasureTime mtime) ->
+      "run-standard " <> timingDistributionBytes d <> " " <>
+      (fromString . map toLower . show $ mtime)
+    RunSimpleExpressions d (MeasureTime mtime) ->
+      "run-simple " <> timingDistributionBytes d <> " " <>
+      (fromString . map toLower . show $ mtime)
+    PollRequestKeys bs (MeasureTime mtime) ->
+      "poll [" <> B8.unwords bs <> "] " <> (fromString . map toLower . show $ mtime)
+    ListenerRequestKey bytestring (MeasureTime mtime) ->
+      "listen " <> bytestring <> " " <> (fromString . map toLower . show $ mtime)
+
+transactionCommandFromText :: MonadThrow m => Text -> m TransactionCommand
+transactionCommandFromText = readTransactionCommandBytes . T.encodeUtf8
+{-# INLINE transactionCommandFromText #-}
+
+readTransactionCommandBytes :: MonadThrow m => B8.ByteString -> m TransactionCommand
+readTransactionCommandBytes = parseBytes "transaction-command" transactionCommandParser
+{-# INLINE readTransactionCommandBytes #-}
+
+transactionCommandParser :: A.Parser TransactionCommand
```

Comment:
Is there a reason you couldn't leverage the JSON instances here instead of having to maintain these parsers?